website/docs/presets/gql-tag-operations.md

```diff
@@ -145,3 +145,140 @@ const Tweets = () => {
+  return (documents as any)[source] ?? {};
+}
+```
+
+However, the `@graphql-codegen/gql-tag-operations-preset` package ships with a babel plugin for rewriting the `gql` usages to actual imports.
+
+This is the code that you write:
+
+```tsx
+import { gql, DocumentType } from '../gql';
+
+const TweetFragment = gql(/* GraphQL */ `
+  fragment TweetFragment on Tweet {
+    id
+    body
+  }
+`);
+```
+
+This is the code after the babel transformation:
+
+```ts
+import { TweetFragmentDoc } from \\"./graphql\\";
+
+const TweetFragment = TweetFragmentDoc
+```
+
+This will result in much smaller bundles when you code-split parts of your code.
+
+Applying the babel plugin is straight-forward:
+
+**.babelrc.mjs**
+
+```ts
+import { babelPlugin } from '@graphql-codegen/gql-tag-operations-preset';
+export default {
+  plugins: [
+    // your other plugins such as typescript syntax stripping etc.
+
+    // make sure the artifactDirectory is the same as your generates path within the codegen.yml
+    [babelPlugin, { artifactDirectory: './src/gql' }],
+  ],
+};
+```
+
+Afterwards, you won't have to worry about bundle size anymore!
+
+## Using module augmentation for using an existing `gql` function from a library
+
+Sometimes the library you are using, let's use `urql` for this example, already exposes a `gql` function and instead of generating a new one you would prefer to just generate types that extend the type-definitions of that `gql` function.
+
+This can be easily achieved by running the `gql-tag-operations` plugin in module augmentation mode!
+
+**codegen.yml**
+
+```yml
+schema: src/path/to/your/schema.graphql
+documents:
+  - 'src/**/*.ts'
+  - '!src/gql/**/*'
+generates:
+  src/gql/:
+    preset: gql-tag-operations-preset
+    presetConfig:
+      # define which module you want to import your gql function from
+      augmentedModuleName: '@urql/core'
+```
+
+When the `augmentedModuleName` option is configured instead of the `./src/gql/index.ts` file a `./src/gql/index.d.ts` file is generated.
+
+```ts
+/* eslint-disable */
+import { TypedDocumentNode as DocumentNode } from '@graphql-typed-document-node/core';
+
+declare module '@urql/core' {
+  export function gql(
+    source: '\n  query Bar {\n    Tweets {\n      ...Lel\n    }\n  }\n'
+  ): typeof import('./graphql').BarDocument;
+  export function gql(source: string): unknown;
+}
+```
+
+Now you can simply use your `gql` function imported from `@urql/core` in your application code for having fully typed document nodes.
+
+```ts
+import { gql } from 'urql';
+
+const FooQuery = gql(/* GraphQL */ `
+  query Foo {
+    Tweets {
+      id
+    }
+  }
+`);
+```
+
+**NOTE**: In case you are using fragments you MUST use the gql-tag-operations plugin as otherwise the `gql` calls that reference global fragments will cause runtime errors, as the `gql` operation cannot find the global fragment.
```

Comment:
```suggestion
**NOTE**: In case you are using fragments you MUST use the gql-tag-operations babel plugin as otherwise, the `gql` calls that reference global fragments will cause runtime errors, as the `gql` operation cannot find the global fragment.
```